src/main/java/teammates/ui/webapi/CreateAccountAction.java

```diff
@@ -43,17 +65,24 @@ public JsonResult execute() throws InvalidHttpRequestBodyException {
             log.severe("Unexpected error", e);
             return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
-        String joinLink = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
-                .withRegistrationKey(instructorList.get(0).getKey())
-                .withEntityType(Const.EntityType.INSTRUCTOR)
-                .toAbsoluteString();
-        EmailWrapper email = emailGenerator.generateNewInstructorAccountJoinEmail(
-                instructorList.get(0).getEmail(), instructorName, joinLink);
-        emailSender.sendEmail(email);
-
-        JoinLinkData output = new JoinLinkData(joinLink);
-        return new JsonResult(output);
+
+        try {
+            logic.joinCourseForInstructor(instructorList.get(0).getKey(), userInfo.id);
+            logic.updateAccountRequest(AccountRequestAttributes
```

Comment:
These two operations should be in separate `try` blocks. We cannot apply a blanket catch-then-rethrow for all the exceptions that are produced here (in general, anywhere within the API layer). For example, neither of the exceptions thrown by the `updateAccountRequest` should be valid here as the entity is guaranteed to exist and valid (the only thing changed is the registration time, which is always valid), thus any of the two being thrown means something is seriously wrong.

Reply:
Thanks for the review! Can I ask since "any of the two being thrown means something is seriously wrong", would it be appropriate to log the error using `log.severe()` here?

Reply:
You can take a look here: https://github.com/TEAMMATES/teammates/blob/4054fef43227e005ae673a3e0e43b4aae2fd358a/src/main/java/teammates/ui/webapi/JoinCourseAction.java#L66-L84